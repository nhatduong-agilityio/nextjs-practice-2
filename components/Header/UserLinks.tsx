import { Button, VStack } from '@chakra-ui/react'
import React from 'react'
import NextLink from 'next/link'

const UserLinks = () => (
  <VStack w='full' gap='20px'>
    <NextLink href='/' passHref>
      Shopping Cart
    </NextLink>
    <NextLink href='/' passHref>
      Wishlist
    </NextLink>
    <NextLink href='/' passHref>
      Sign Up
    </NextLink>
    <NextLink href='/' passHref style={{ width: '100%' }}>
      <Button
        borderRadius='10px'
        width='full'
        variant='secondary'
        height={['52px', '66px']}
        fontSize={['sm', 'md']}
      >
        Login
      </Button>
    </NextLink>
  </VStack>
)

export default UserLinks
