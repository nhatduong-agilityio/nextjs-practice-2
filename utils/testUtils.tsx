import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
