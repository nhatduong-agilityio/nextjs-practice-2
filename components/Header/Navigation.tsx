import { Flex, Button, HStack } from '@chakra-ui/react'
import React, { memo } from 'react'
import NextLink from 'next/link'

// Components
import CartIcon from '@components/Icons/CartIcon'
import WishlistIcon from '@components/Icons/WishlistIcon'
import UserLinks from '@components/Header/UserLinks'

interface IProps {
  toggleMenu: boolean
}

const Navigation = memo(({ toggleMenu }: IProps) => (
  <>
    <HStack style={{ marginTop: '0' }} spacing='50px' display={['none', 'flex']}>
      <NextLink href='/' passHref aria-label='cart'>
        <CartIcon
          width='24px'
          height='24px'
          viewBox='0 0 24 24'
          fill='black'
          stroke='black'
          strokeWidth='0.25'
        />
      </NextLink>
      <NextLink href='/' passHref aria-label='wishlist'>
        <WishlistIcon width='24px' height='22px' viewBox='0 0 24 22' fill='black' stroke='black' />
      </NextLink>
      <HStack style={{ marginTop: '0' }} spacing='30px'>
        <NextLink
          href='/'
          passHref
          style={{
            width: '100%',
          }}
        >
          <Button
            width='full'
            variant='primary'
            height={['52px', '60px']}
            fontSize={['sm', 'md']}
            padding='0'
          >
            Sign Up
          </Button>
        </NextLink>
        <NextLink
          href='/'
          passHref
          style={{
            width: '100%',
          }}
        >
          <Button
            borderRadius='10px'
            width='full'
            minWidth='123px'
            variant='secondary'
            height={['52px', '60px']}
            fontSize={['sm', 'md']}
          >
            Login
          </Button>
        </NextLink>
      </HStack>
    </HStack>
    <Flex
      w='full'
      display={[
        `${!toggleMenu ? 'none' : 'flex'}`,
        `${!toggleMenu ? 'none' : 'flex'}`,
        'none',
        'none',
      ]}
      bgColor='white'
      h='100vh'
      pos='fixed'
      top='77px'
      left='0'
      overflowY='auto'
      flexDir='column'
      borderTop='1px solid black'
      p={5}
      zIndex='100'
    >
      <UserLinks />
    </Flex>
  </>
))

Navigation.displayName = 'Navigation'

export default Navigation
