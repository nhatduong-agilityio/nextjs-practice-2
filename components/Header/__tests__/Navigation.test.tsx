// Utils
import { render, waitFor } from '@utils/testUtils'

// Components
import Navigation from '../Navigation'
import UserLinks from '../UserLinks'

describe('Navigation render', () => {
  const propsMock = {
    toggleMenu: true,
  }

  it('Should Navigation match Snapshot', () => {
    const { container } = render(<Navigation {...propsMock} />)

    expect(container).toMatchSnapshot()
  })

  it('Should user links show when opened', async () => {
    render(<Navigation {...propsMock} />)

    const { container } = render(<UserLinks />)

    let lazyContainer

    await waitFor(() => {
      lazyContainer = container
    })

    expect(lazyContainer).toMatchSnapshot()
  })
})
