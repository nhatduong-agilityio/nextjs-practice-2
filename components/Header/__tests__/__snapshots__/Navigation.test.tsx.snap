// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render Should Navigation match Snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-17fre3w"
    style="margin-top: 0px;"
  >
    <a
      href="/"
    >
      <svg
        class="chakra-icon css-svtcwa"
        focusable="false"
        stroke-width="0.25"
        viewBox="0 0 24 24"
      >
        <g
          clip-path="url(#clip0_37_488)"
        >
          <path
            d="M23.9584 4.04031L23.9584 4.04025C23.802 3.8315 23.5572 3.70833 23.2969 3.70833H6.2114L5.73041 1.52775L5.7304 1.52771C5.6463 1.14697 5.31075 0.875 4.92188 0.875H0.703126C0.244855 0.875 -0.125 1.24893 -0.125 1.70833C-0.125 2.16774 0.244855 2.54167 0.703126 2.54167H4.25748L6.75284 13.8543C6.01168 14.2208 5.50001 14.9887 5.50001 15.875C5.50001 17.1149 6.5014 18.125 7.73439 18.125H20.4844C20.9427 18.125 21.3125 17.7511 21.3125 17.2917C21.3125 16.8323 20.9427 16.4583 20.4844 16.4583H7.73439C7.41666 16.4583 7.15626 16.1973 7.15626 15.875C7.15626 15.5531 7.41571 15.2926 7.73285 15.2919H7.73366C7.73633 15.2919 7.7387 15.2918 7.74064 15.2917H20.4844C20.8547 15.2917 21.1793 15.0443 21.2807 14.6871L24.0932 4.77038L23.9729 4.73627L24.0932 4.77037C24.1644 4.51917 24.1145 4.24903 23.9584 4.04031ZM8.39879 13.625L6.57893 5.375H22.1993L19.8595 13.625H8.39879Z"
          />
          <path
            d="M6.875 21C6.875 22.1718 7.82819 23.125 9 23.125C10.1718 23.125 11.125 22.1718 11.125 21C11.125 19.8282 10.1718 18.875 9 18.875C7.82819 18.875 6.875 19.8282 6.875 21ZM9 20.4583C9.2985 20.4583 9.54167 20.7015 9.54167 21C9.54167 21.2985 9.2985 21.5417 9 21.5417C8.7015 21.5417 8.45833 21.2985 8.45833 21C8.45833 20.7015 8.7015 20.4583 9 20.4583Z"
          />
          <path
            d="M16.875 21C16.875 22.1718 17.8282 23.125 19 23.125C20.1718 23.125 21.125 22.1718 21.125 21C21.125 19.8282 20.1718 18.875 19 18.875C17.8282 18.875 16.875 19.8282 16.875 21ZM19 20.4583C19.2985 20.4583 19.5417 20.7015 19.5417 21C19.5417 21.2985 19.2985 21.5417 19 21.5417C18.7015 21.5417 18.4583 21.2985 18.4583 21C18.4583 20.7015 18.7015 20.4583 19 20.4583Z"
          />
        </g>
        <defs>
          <clippath
            id="clip0_37_488"
          >
            <rect
              fill="white"
              height="24"
              width="24"
            />
          </clippath>
        </defs>
      </svg>
    </a>
    <a
      href="/"
    >
      <svg
        class="chakra-icon css-1y3afmn"
        focusable="false"
        viewBox="0 0 24 22"
      >
        <path
          d="M3.80396 3.95029C4.66754 2.99331 5.85332 2.46685 7.14587 2.46685C8.08863 2.46685 8.95393 2.77234 9.72114 3.37853C10.4075 3.92098 10.8884 4.60954 11.1717 5.09477L11.1717 5.0948C11.3464 5.39386 11.6564 5.57476 12 5.57476C12.3436 5.57476 12.6536 5.39386 12.8283 5.0948L12.7061 5.02342L12.8283 5.0948C13.1117 4.60954 13.5926 3.92096 14.2789 3.37851C15.0461 2.77234 15.9114 2.46685 16.8543 2.46685C18.1467 2.46685 19.3326 2.99332 20.196 3.95027C21.0735 4.92282 21.5614 6.27274 21.5614 7.75591C21.5614 9.31145 20.9982 10.7092 19.7095 12.2982C18.4592 13.8399 16.5962 15.4625 14.4222 17.3559C13.69 17.9932 12.8635 18.7131 11.9983 19.4817C11.1381 18.7145 10.3129 17.9958 9.58233 17.3596L9.58105 17.3585L9.57829 17.3562C9.5781 17.356 9.57792 17.3559 9.57773 17.3557C7.40381 15.4626 5.54088 13.8397 4.29065 12.2982L4.17415 12.3927L4.29065 12.2982C3.00185 10.7092 2.43856 9.31145 2.43856 7.75591C2.43856 6.27274 2.92652 4.92283 3.80396 3.95029ZM3.80396 3.95029L3.6926 3.84979L3.80396 3.95029ZM8.54831 18.587L8.54943 18.588C9.30548 19.2463 10.1616 19.9919 11.0496 20.7853C11.3121 21.0203 11.6496 21.15 12 21.15C12.3502 21.15 12.6879 21.0203 12.9503 20.7856C13.8402 19.9904 14.6977 19.2436 15.4551 18.5842L15.4551 18.5842L15.4642 18.5763C17.6731 16.6527 19.5948 14.9792 20.9347 13.3272C22.4396 11.4718 23.15 9.69919 23.15 7.75591C23.15 5.87564 22.5191 4.13455 21.3663 2.85678L21.2555 2.95672L21.3663 2.85677C20.199 1.56314 18.5957 0.85 16.8543 0.85C15.5496 0.85 14.354 1.27226 13.3047 2.10157C12.8264 2.47949 12.3899 2.93304 12 3.45527C11.6102 2.93302 11.1736 2.47948 10.6955 2.10158L10.6955 2.10157C9.64612 1.27226 8.45055 0.85 7.14587 0.85C5.40427 0.85 3.80116 1.56314 2.6339 2.85677L2.6339 2.85678C1.48107 4.13455 0.85 5.87563 0.85 7.75591C0.85 9.6992 1.56061 11.4718 3.06551 13.3274L3.18201 13.2329L3.06551 13.3274C4.4057 14.9797 6.32824 16.6538 8.53825 18.5782L8.54445 18.5836L8.54557 18.5846L8.54831 18.587Z"
          stroke-width="0.3"
        />
      </svg>
    </a>
    <div
      class="chakra-stack css-vlp7kp"
      style="margin-top: 0px;"
    >
      <a
        href="/"
        style="width: 100%;"
      >
        <button
          class="chakra-button css-7as66v"
          type="button"
        >
          Sign Up
        </button>
      </a>
      <a
        href="/"
        style="width: 100%;"
      >
        <button
          class="chakra-button css-1jw3zmz"
          type="button"
        >
          Login
        </button>
      </a>
    </div>
  </div>
  <div
    class="css-1whv18x"
  >
    <div
      class="chakra-stack css-1trjysb"
    >
      <a
        href="/"
      >
        Shopping Cart
      </a>
      <a
        href="/"
      >
        Wishlist
      </a>
      <a
        href="/"
      >
        Sign Up
      </a>
      <a
        href="/"
        style="width: 100%;"
      >
        <button
          class="chakra-button css-1b61qg5"
          type="button"
        >
          Login
        </button>
      </a>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`Navigation render Should user links show when opened 1`] = `
<div>
  <div
    class="chakra-stack css-1trjysb"
  >
    <a
      href="/"
    >
      Shopping Cart
    </a>
    <a
      href="/"
    >
      Wishlist
    </a>
    <a
      href="/"
    >
      Sign Up
    </a>
    <a
      href="/"
      style="width: 100%;"
    >
      <button
        class="chakra-button css-1b61qg5"
        type="button"
      >
        Login
      </button>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
